components/webapp-mgt/org.wso2.carbon.webapp.mgt/src/main/java/org/wso2/carbon/webapp/mgt/sso/SAMLSSOValve.java

```diff
@@ -237,6 +240,29 @@ else if (request.getRequestURI().endsWith(
                             Boolean.parseBoolean(ssoSPConfigProperties.getProperty(
                                     WebappSSOConstants.HANDLE_CONSUMER_URL_AFTER_SLO))) {
 
+                        if (Boolean.valueOf(ssoSPConfigProperties
+                                .getProperty(WebappSSOConstants.ENABLE_IDP_SESSION_VALIDATION_BEFORE_LOGOUT, "false"))
+                                && ssoAgentConfig.getSAML2().isPassiveAuthn()) {
+
+                            String saml2ResponseString = new String(Base64.decode(
+                                    request.getParameter(SSOAgentConstants.SAML2SSO.HTTP_POST_PARAM_SAML2_RESP)),
+                                    Charset.forName("UTF-8"));
+                            org.opensaml.saml2.core.Response saml2Response = (org.opensaml.saml2.core.Response) SSOAgentUtils
+                                    .unmarshall(saml2ResponseString);
+                            String htmlPayload;
+                            ssoAgentConfig.getSAML2().setPassiveAuthn(false);
+
+                            if (isNoPassive(saml2Response)) {
```

Comment:
isNotPassive?

Reply:
This is to verify the SAML StatusCode: `urn:oasis:names:tc:SAML:2.0:status:NoPassive` in SAML response. I guess `isNoPassive` would be ok with this?